python/rikai/spark/functions/vision.py

```diff
@@ -133,3 +141,37 @@ def video_to_images(
         )
         for idx, img in enumerate(video_iterator)
     ]
+
+
+@udf(returnType=ImageType())
+def spectrogram_image(video) -> Image:
+    """Applied ffmpeg filter to generate spectrogram
+    Parameters
+    ----------
+    video : Video
+        A video object, either YouTubeVideo or VideoStream.
+    Return
+    ------
+    Image
+        Return an Image of the audio spectrogram
+    """
+    assert isinstance(video, YouTubeVideo) or isinstance(
+        video, VideoStream
+    ), "Input type must be YouTubeVideo or VideoStream"
+
+    base_path = video.vid if isinstance(video, YouTubeVideo) else video.uri
+    video_uri = (
+        video.get_stream().uri
+        if isinstance(video, YouTubeVideo)
+        else video.uri
+    )
+    output, _ = (
+        ffmpeg.input(video_uri)
+        .filter("showspectrumpic", "224x224", legend=0)
```

Comment:
Are these filters configurable later?

Reply:
Added a parameter to set default size.
The time dimension of the spectrogram will scale with video duration and can be resized or trimmed.

Reply:
Added a segment parameter to help configure time range.